test/common/config/registry_test.cc

```diff
@@ -103,6 +103,19 @@ TEST(RegistryTest, DEPRECATED_FEATURE_TEST(WithDeprecatedFactoryPublished)) {
                           ->name());
 }
 
+class TestWithDeprecatedPublishedFactoryNoDeprecatedNames : public PublishedFactory {
+public:
+  std::string name() const override { return ""; }
+};
+
+TEST(RegistryTest, DEPRECATED_FEATURE_TEST(AssertsIfNoDeprecatedNameGiven)) {
+  // Expects an assert to raise if we register a factory that has an empty name
+  // and no associated deprecated names.
+  EXPECT_DEATH((Registry::RegisterFactory<TestWithDeprecatedPublishedFactoryNoDeprecatedNames,
+                                          PublishedFactory>({})),
```

Comment:
Are you adding test coverage for a branch that has an ASSERT or a RELEASE_ASSERT?  If the former, I think you need EXPECT_DEBUG_DEATH instead of EXPECT_DEATH, otherwise this test will fail when you run:
bazel test -c opt //test/common/config:registry_test